client/src/lib/CKANApi.js

```diff
@@ -146,49 +146,30 @@ class CKANApi {
    * @returns {Promise}
    */
   static loadDataset(endpoint, dataset) {
-    // Define a function to handle the response from the API (for re-use)
-    const handleResponse = json => {
-      // Vet out the response being invalid or empty
-      if (!json.success || !json.result.count) {
-        return false;
-      }
-
-      // Pull the first result
-      const result = json.result.results[0];
-
-      // Ensure the name or ID matches as we're looking for an exact
-      if (result.name !== dataset && result.id !== dataset) {
-        return false;
-      }
-
-      // Return the dataset
-      return result;
-    };
-
-    // TODO Replace with package_show once data.govt.nz fixes their bug
-    // package_show makes sense here but it doesn't actually work with data.govt.nz because of CORS
-
     // We look for a dataset (package) matching first the name, and then the ID if nothing is found
-    return this.makeRequest(endpoint, 'package_search', {
-      fq: `name:${dataset}`,
+    return this.makeRequest(endpoint, 'package_show', {
+      id: dataset,
```

Comment:
Could you please update the comment on `L149` (or add one on here) to explain that the `id:` property can be either a name or an ID, and CKAN handles searching both internally. Otherwise it looks like there's an omission in the code path.

Reply:
Updated :) : https://github.com/silverstripe/silverstripe-ckan-registry/compare/11fc6a32167f3a5ed1d2a9493ef5b7d6899273a0..d5f2ded4d90fa1bd77919b0cc45eb4ac0fccbe54